src/p2p/net_node.inl

```diff
@@ -1095,19 +1161,22 @@ namespace nodetool
     {
       if(conn_count < expected_white_connections)
       {
-        //start from white list
-        if(!make_expected_connections_count(true, expected_white_connections))
+        //start from anchor list
+        if(!make_expected_connections_count(anchor, P2P_DEFAULT_ANCHOR_CONNECTIONS_COUNT))
           return false;
-        //and then do grey list
-        if(!make_expected_connections_count(false, m_config.m_net_config.connections_count))
+        //then do white list
+        if(!make_expected_connections_count(white, expected_white_connections))
```

Comment:
Anchors can be considered a subset of white peers with higher priority of connection upon restart.

Potentially only 2 of those 8 would be used, the rest would be from the white list (1000 max) and gray list (5000 max).

Instead of picking 5 random peers from the white list, the system will try to connect to at least to 2 from the anchor, if those connections are not successful it continue picking nodes from the white list.

The purpose is to keep some connections to old known nodes to prevent the eclipse attack.